fix for #10327 : replacing support for get_ipython().magic()

adityausathe opened the pull request:
There was no direct calls made to `get_ipython().run_line_magic()`, instead we were calling `get_ipython().magic()` first which calls `get_ipython().run_line_magic()` from it's body. That was the reason `hist -t` showing `get_ipython().magic()` instead of `get_ipython().run_line_magic()` because that's what was happening in there.

I replaced calls to `get_ipython().magic()` by `get_ipython().run_line_magic()`. But, there are some tests in `IPython/core/tests/..` which still make calls to `get_ipython().magic()`. I haven't changed them.
#10327

adityausathe committed:
fix for #10327 : get_ipython.magic() replaced with get_ipython.run_line_magic()

adityausathe committed:
fix for #10327 : get_ipython().magic() replaced with get_ipython().run_line_magic()

adityausathe commented:
It seems that the tests should also need to be changed in order to accommodate the changes. The checks are failing at those points. Are we changing the test cases?
@Carreau

adityausathe committed:
fix for #10327 : get_ipython().magic() replaced with get_ipython().run_line_magic()

Carreau commented:
Hi there, thanks you probably want to synchronize with #10612 which is also trying to update this. 
Yes the test need to be updated, but we need to make sure `get_ipython().magic()` still works at least until it is removed.

adityausathe committed:
Fix #10327: Test fails handled and magic() backward compatibilty added

adityausathe committed:
Fix #10327: Test fails handled and magic() backward compatibilty added

adityausathe commented:
Summary: 
- Replaced direct calls to `get_ipython().magic()` with `get_ipython().run_line_magic()`
- Updated test cases to accommodate the change.
- Added backward compatibility support for `get_ipython().magic()`, by updating `stack_depth`

@Carreau 
@takluyver

srinivasreddy commented on IPython/core/inputtransformer.py:
```diff
@@ -198,11 +198,14 @@ def _make_help_call(target, esc, lspace, next_input=None):
                 else 'psearch' if '*' in target \
                 else 'pinfo'
     arg = " ".join([method, target])
```
this line can be removed.

srinivasreddy reviewed (approved):
LGTM. thanks.

takluyver commented on IPython/core/interactiveshell.py:
```diff
@@ -2044,7 +2044,7 @@ def register_magic_function(self, func, magic_kind='line', magic_name=None):
                                   magic_kind=magic_kind, magic_name=magic_name)
 
-    def run_line_magic(self, magic_name, line):
+    def run_line_magic(self, magic_name, line, back_compat=False):
```
Maybe make the parameter `_stack_depth=1` rather than `back_compat=False` - that's a bit clearer about what it means. The leading underscore would suggest that it's not for normal use.

takluyver commented on IPython/core/tests/test_inputtransformer.py:
```diff
@@ -57,13 +57,13 @@ def transform_checker(tests, transformer, **kwargs):
-       [(u'a =% who', "a = get_ipython().magic({u}'who')"),
-        (u'b = %who', "b = get_ipython().magic({u}'who')"),
-        (u'c= %ls', "c = get_ipython().magic({u}'ls')"),
+       [(u'a =% who', "a = get_ipython().run_line_magic({u}'who', {u}'')"),
```
So long as we're targeting IPython 6.x, we can get rid of the `{u}` things in the tests - that's for Python 2 compatibility, which we've dropped now.

adityausathe committed:
fix for #10327: _stack_depth added and {u} string literals removed from core/tests

adityausathe commented:
@takluyver
Thanks for the feedback. 
Incorporated suggested changes.

takluyver commented:
Thanks
